File: treinamentoP1/combinacao0.2.c
#include <stdio.h>

double comb(double n, double p);

int main(void)
{
	double n,p;
	scanf("%lf %lf", &n, &p);
	printf("%.0f\n", comb(n,p));
	
	return 0;
}

double comb(double n, double p)
{
	if( p == 0)
		return 1;
	else{
		return comb(n-1,p-1) * n / p *  (n-p) * (n-(p-1)) ;
	}
}

File: treinamentoP1/combinacao0.2.py
# Solução teoricamente possivel Mas,
# inviavel na pratica


def comb(n,k):
	if(k==0):
		return 1
	else:
		return comb(n-1,k-1) * (n / (k * (n-k) * (n-k-1)) )
		
print( comb(10,3) )
